src/autoscaler/api/publicapiserver/oauth.go

```diff
@@ -64,18 +60,24 @@ func (oam *OAuthMiddleware) Middleware(next http.Handler) http.Handler {
 
 		isUserSpaceDeveloper, err := oam.cfClient.IsUserSpaceDeveloper(userToken, appId)
 		if err != nil {
-			oam.logger.Error("failed to check spacedeveloper permissions", err, nil)
-			handlers.WriteJSONResponse(w, http.StatusInternalServerError, models.ErrorResponse{
-				Code:    "Interal-Server-Error",
-				Message: "Failed to check space developer permission"})
-			return
+			if err == cf.ErrUnauthrorized {
+				http.Error(w, "{}", http.StatusUnauthorized)
+				return
+			} else {
+				oam.logger.Error("failed to check spacedeveloper permissions", err, nil)
```

Comment:
spacedeveloper ==> space developer

Reply:
I suggest to validate the `usertoken` first prior to `IsUserAdmin` check to return 401 error as early as possible if the user token is wrong